transform-plugins/src/main/java/co/cask/hydrator/plugin/Decoder.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.Schema.Field;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import org.apache.commons.codec.binary.Base32;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Decodes the input fields as BASE64, BASE32 or HEX.
+ * 
+ * Please note that Encoder and Decoder might look the same right now, but 
+ * in near future they will diverge.
+ */
+@Plugin(type = "transform")
+@Name("Decoder")
+@Description("Decodes the input field(s) using Base64, Base32, or Hex")
+public final class Decoder extends Transform<StructuredRecord, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(Decoder.class);
+  private final Config config;
+
+  // Output Schema associated with transform output.
+  private Schema outSchema;
+
+  // Mapping of input field to decoder type.
+  private final Map<String, DecoderType> decodeMap = new TreeMap<>();
+
+  // Decoder handlers.
+  private final Base64 base64Decoder = new Base64();
+  private final Base32 base32Decoder = new Base32();
+  private final Hex hexDecoder = new Hex();
+
+  // Output Field name to type map
+  private final Map<String, Schema.Type> outSchemaMap = new HashMap<>();
+
+  // This is used only for tests, otherwise this is being injected by the ingestion framework.
+  public Decoder(Config config) {
+    this.config = config;
+  }
+
+  private void parseConfiguration(String config) throws IllegalArgumentException {
+    String[] mappings = config.split(",");
```

Comment:
Using the `com.google.common.base.Splitter` class would simplify this method, especially since it handles the error cases, such as incorrect formatting (missing a colon in one of the mappings) or duplicate entries.

```java
Map<String, String> inputConfig = Splitter.on(",").withKeyValueSeparator(":").split(config);
for (Map.Entry<String, String> entry : inputConfig.entrySet()) {
  DecoderType decoderType = DecoderType.valueOf(entry.getValue().toUpperCase());
  decodeMap.put(entry.getKey(), decoderType);
}
```